volume-providers/vpc/provider/wait_for_detach_volume.go

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************************
+ * IBM Confidential
+ * OCO Source Materials
+ * IBM Cloud Container Service, 5737-D43
+ * (C) Copyright IBM Corp. 2018, 2019 All Rights Reserved.
+ * The source code for this program is not  published or otherwise divested of
+ * its trade secrets, irrespective of what has been deposited with
+ * the U.S. Copyright Office.
+ ******************************************************************************/
+
+package provider
+
+import (
+	"github.com/IBM/ibmcloud-storage-volume-lib/lib/provider"
+	util "github.com/IBM/ibmcloud-storage-volume-lib/lib/utils"
+	userError "github.com/IBM/ibmcloud-storage-volume-lib/volume-providers/vpc/messages"
+	"go.uber.org/zap"
+	"time"
+)
+
+// WaitForDetachVolume waits for volume to be detached from node. e.g waits till no volume attachment is found
+func (vpcs *VPCSession) WaitForDetachVolume(volumeAttachmentTemplate provider.VolumeAttachmentRequest) error {
+	vpcs.Logger.Debug("Entry of WaitForDetachVolume method...")
+	defer vpcs.Logger.Debug("Exit from WaitForDetachVolume method...")
+	var err error
+	vpcs.Logger.Info("Validating basic inputs for WaitForDetachVolume method...", zap.Reflect("volumeAttachmentTemplate", volumeAttachmentTemplate))
+	err = vpcs.validateAttachVolumeRequest(volumeAttachmentTemplate)
+	if err != nil {
+		return err
+	}
+	maxTimeout, maxRetryAttempt, retryGapDuration := vpcs.Config.GetTimeOutParameters()
+	retryCount := 0
+	vpcs.Logger.Info("Waiting for volume to be detached", zap.Int("maxTimeout", maxTimeout))
+	for retryCount < maxRetryAttempt {
```

Comment:
Can we have this retry logic in some wrapper method similar to what we have for other calls

Reply:
Agree, But lets do that in separate PR . We will have to move retry logic from provider to lib/interface package so that it will be used for all the providers